discovery_engine/lib/src/ffi/types/document/document.dart

```diff
@@ -60,6 +67,12 @@ class DocumentFfi with EquatableMixin {
     resource.writeNative(ffi.document_place_of_resource(place));
   }
 
-  @override
-  List<Object?> get props => [id, stackId, smbertEmbedding, resource];
+  Document toDocument({required int batchIndex}) {
+    return Document(
+      documentId: id,
+      stackId: stackId,
+      resource: resource,
+      batchIndex: batchIndex,
+    );
+  }
```

Comment:
since this is a one-liner anyways, it could just be written as
```suggestion
  Document toDocument({required int batchIndex}) => Document(
        documentId: id,
        stackId: stackId,
        resource: resource,
        batchIndex: batchIndex,
      );
```

Reply:
I know, I initially did. But the formatting gives me the creeps in this case. Not sure why. I can change it if necessary.